torch/csrc/jit/tensorexpr/ir_deserializer.cpp

```diff
@@ -0,0 +1,345 @@
+  SyncThreads* syncthreads = new SyncThreads();
+  return syncthreads;
+}
+
+Let* IRDeserializer::deserializeLet(const json& json) {
+  const Var* var = dynamic_cast<const Var*>(dsExpr(json["var"]));
+  Let* let = new Let(var, dsExpr(json["value"]));
+  return let;
+}
+
+Store* IRDeserializer::deserializeStore(const json& json) {
+  const Buf* buf = dynamic_cast<const Buf*>(dsExpr(json["buf"]));
+  Store* store = new Store(
+      buf,
+      dsVecExpr(json["indices"]),
+      dsExpr(json["value"]),
+      dsExpr(json["mask"]));
+  return store;
+}
+
+Block* IRDeserializer::deserializeBlock(const json& json) {
+  Block* block = new Block(dsVecStmt(json["stmts"]));
+  return block;
+}
+
+For* IRDeserializer::deserializeFor(const json& v) {
+  const Var* var = static_cast<const Var*>(dsExpr(v["var"]));
+  const json& loop_options = v["loop_options"];
+  const json& buffer_mapping_json = loop_options["buffer_mapping"];
+  std::unordered_map<std::string, const Buf*> buffer_mapping;
+  for (const auto& obj : buffer_mapping_json.items()) {
+    std::string key = obj.key();
+    const Buf* buf = dynamic_cast<const Buf*>(dsExpr(obj.value()));
+    buffer_mapping[key] = buf;
+  }
+  LoopOptions lo;
+  lo.set_buffer_mapping(buffer_mapping);
+  lo.set_gpu_thread_index(loop_options["gpu_thread_index"]);
+  lo.set_gpu_block_index(loop_options["gpu_block_index"]);
+  For* for_var = new For(
+      var, dsExpr(v["start"]), dsExpr(v["stop"]), dsStmt(v["body"]), lo);
+  return for_var;
+}
+
+Free* IRDeserializer::deserializeFree(const json& json) {
+  const Var* var = static_cast<const Var*>(dsExpr(json["buffer_var"]));
+  Free* free = new Free(var);
+  return free;
+}
+
+Cond* IRDeserializer::deserializeCond(const json& json) {
+  Cond* cond = new Cond(
+      dsExpr(json["condition"]),
+      dsStmt(json["true_stmt"]),
+      dsStmt(json["false_stmt"]));
+  return cond;
+}
+
+const Expr* IRDeserializer::dsExprImpl(const json& v) {
+  std::string expr_type = v["expr_type"];
+
+#define CHECK_BINARY_OP(cond)                           \
+  if (expr_type == #cond) {                             \
+    return deserializeBinaryOp(v, IRNodeType::k##cond); \
+  }
+
+  CHECK_BINARY_OP(Xor)
+  CHECK_BINARY_OP(Add)
+  CHECK_BINARY_OP(Sub)
+  CHECK_BINARY_OP(Mul)
+  CHECK_BINARY_OP(Div)
+  CHECK_BINARY_OP(Mod)
+  CHECK_BINARY_OP(Max)
+  CHECK_BINARY_OP(Min)
+  CHECK_BINARY_OP(And)
+  CHECK_BINARY_OP(Or)
+  CHECK_BINARY_OP(Xor)
+  CHECK_BINARY_OP(Lshift)
+  CHECK_BINARY_OP(Rshift)
+  CHECK_BINARY_OP(CompareSelect)
+  CHECK_BINARY_OP(RoundOff)
+#undef CHECK_BINARY_OP
+
+#define CHECK_EXPR_TYPE(expr_name)    \
+  if (expr_type == #expr_name) {      \
+    return deserialize##expr_name(v); \
+  }
+  CHECK_EXPR_TYPE(Cast);
+  CHECK_EXPR_TYPE(Var);
+  CHECK_EXPR_TYPE(Ramp);
+  CHECK_EXPR_TYPE(Broadcast);
+  CHECK_EXPR_TYPE(IfThenElse);
+  // CHECK_EXPR_TYPE(Term);
+  // CHECK_EXPR_TYPE(Polynomial);
+  CHECK_EXPR_TYPE(MaxTerm);
```

Comment:
MinTerm and MaxTerm have the same lifetime as Term and Polynomial (ie. they exist only during arithmetic simplification) so if you're not including Term and Poly (which sounds right to me) then I wouldn't include these as well.

Min and Max (binary ops) are important though.